ipa-core/src/protocol/ipa_prf/malicious_security/quadratic_proofs.rs

```diff
@@ -0,0 +1,933 @@
+use futures::StreamExt;
+use futures_util::{stream, TryStreamExt};
+use ipa_macros::Step;
+
+use crate::{
+    error::Error,
+    ff::{Field, Invert},
+    helpers::{Direction, ReceivingEnd, SendingEnd},
+    protocol::{
+        basics::validate_replicated_shares::validate_sum_to_zero,
+        context::Context,
+        ipa_prf::malicious_security::{
+            fiat_shamir::{fiat_shamir_prover, fiat_shamir_verifier},
+            lagrange::{
+                compute_lagrange_base, compute_lagrange_denominator, generate_evaluation_points,
+                lagrange_evaluation, lagrange_evaluation_precomputed,
+            },
+        },
+        prss::SharedRandomness,
+        RecordId,
+    },
+    seq_join::seq_join,
+};
+
+#[derive(Step)]
+pub(crate) enum Step {
+    HashFromLeft,
+    HashFromRight,
+    PQFromLeft,
+    TwoOutOfTwoZeroCheck,
+}
+
+/// Non-Interactive Distributed Zero Knowledge Proof
+/// a `NIDZKP` which is computed over multiple recursions , is a vector of `g` polynomials
+/// the length is the amount of proof recursions
+/// `g` is secret shared using PRSS: left share is stored in struct, right share is generated via PRSS
+/// the length of the left share, `g_left`, determines the degree of the polynomial
+/// for a low communication rate, i.e. `log (amount of multiplications)`,
+/// the length needs to be constant in the amount of multiplications
+/// length might be different across recursions (therefore used `Vec` instead of `GenericArray`)
```

Comment:
How are these two sentences not in conflict with one another? I understand that the degree of the polynomial can vary in each recursion, so what do you mean by "the length needs to be constant in the amount of multiplications"? What does that mean?